File: backend/src/main/java/com/maciej/poreba/backend/application/payload/response/CarResponse.java
package com.maciej.poreba.backend.application.payload.response;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.maciej.poreba.backend.application.model.Car;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Builder
@AllArgsConstructor
@NoArgsConstructor
@Data
public class CarResponse {
   private String model;
   private String brand;
   private String yearOfProduction;
   private double price;
   private String registrationNumber;
   private boolean available;
   private Date availableDate;

   public CarResponse(Car car){
       this.model = car.getModel();
       this.brand = car.getBrand();
       this.yearOfProduction = car.getYearOfProduction();
       this.price = car.getPrice();
       this.registrationNumber = car.getRegistrationNumber();
       this.available = car.isAvailable();
       this.availableDate = car.getReservedUntil();
   }
}

File: frontend/src/app/application/models/CarResponse.ts
export class CarResponse {
  model: string
  brand: string
  yearOfProduction: Date
  price: number
  registrationNumber: string
  available: boolean
  availableDate: Date


  constructor(model: string = "",
              brand: string = "",
              yearOfProduction: Date = new Date(),
              price: number = 0,
              registrationNumber: string = "",
              available: boolean = true,
              availableDate: Date = new Date()) {
    this.model = model;
    this.brand = brand;
    this.yearOfProduction = yearOfProduction;
    this.price = price;
    this.registrationNumber = registrationNumber;
    this.available = available;
    this.availableDate = availableDate;
  }

  toString(){
    return "model: " + this.model + "\n" +
     "brand" + this.brand + "\n" +
     "yearOfProduction" + this.yearOfProduction + "\n" +
     "price" + this.price + "\n" +
     "registrationNumber" + this.registrationNumber + "\n" +
     "available" + this.available + "\n" +
     "availableDate" + this.availableDate
  }
}
